### DISCUSSION TITLE
Music Visualiser option during full screen music playback.

### DISCUSSION THREAD

**Discussion by flip-dots** [jellyfin/jellyfin-vue#2011] [Features]
I thought it would be cool if there was a music visualiser as an option when you full screen music so I threw together a quick prototype to see if it was possible and I think the results are pretty cool.

![image](https://github.com/jellyfin/jellyfin-vue/assets/42912136/d2d39344-e695-4f9a-840c-1b8ba42958e2)

I think the library I used uses a compatible [licence](https://audiomotion.dev/#/?id=license) so I think it should be ok on that front, and if you want to see some of the possibilities there is a [demo](https://audiomotion.dev/demo/fluid.html) site for the library.

So before I spend much more than an afternoon on it, if done correctly is this a feature that you guys would accept?

**Comment by ferferga** [comment]
I think stuff like this has come up already in the past, and I don't think we ever had anything against it. 

We didn't ever work on it since lately we're basically all in with refactors and stabilise the code, but it's something I (as a primary music user) myself wanted. We want to make an adaptable app for every media type as listed in our roadmap, and we have a fullscreen music page we can enrich, so why not? :P

I prefer more subtle effects (I'm more of  background colors vibrating) but that's probably something we can let users configure once we have some settings pages done (wanna help with those too after this? :P). For now on, I think a button in the app bar at the left that lets you toggle between the current album view and this is fine.

As long as you do a good implementation (this is what the PR review process is for) and the library you use can be extended for other effects (so we can expand them in the future to cater different tastes. If not, please find others, I did my research back in the day and there are a lot of these libraries) you have my ✅.

Thoughts @ThibaultNocchi ?

**Comment by ferferga** [comment]
@flip-dots Any update about this?

**Reply by flip-dots** [reply]
I cleaned up the code a while ago and it seems to work ok, but I haven't submitted a pull request yet because I don't currently have the time to fix my inevitable mistakes due to university exams. I will be free of those in a week, so I plan on dealing with it all then. If i'm not super busy this summer I might be able to help out with the setting pages but I don't control my procrastination demon, so no promises.  :)

**Reply by ferferga** [reply]
@flip-dots No worries, I'm in the same situation as you, so I understand perfectly. Took the plunge today to do some cleanup of dependency updates, but now I'm back to study :).

Just wanted to be sure that we had you on board and you didn't had any problem 🙌!

Wish you luck for your exams!

**Comment by ferferga** [comment]
Added in #2043
